plugins/web-searcher/PluginEndpoint.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using PluginShared;
+using WebSearcher.Models;
+
+namespace WebSearcher;
+
+/// <summary>
+/// Plugin endpoints
+/// </summary>
+public class PluginEndpoint
+{
+    private readonly PluginConfig _config;
+    private readonly ILogger _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PluginEndpoint"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    public PluginEndpoint(PluginConfig config, ILogger<PluginEndpoint> logger)
+    {
+        this._config = config;
+        this._logger = logger;
+    }
+
+    /// <summary>
+    /// Gets the plugin manifest.
+    /// </summary>
+    /// <param name="req">The http request data.</param>
+    /// <returns>The manifest in Json</returns>
+    [Function("WellKnownAIPlugin")]
+    public async Task<HttpResponseData> WellKnownAIPlugin(
```

Comment:
nit: Makes more clear to what we're requesting (the manifest rather than the plugin itself)
```suggestion
    public async Task<HttpResponseData> WellKnownAIPluginManifest(
```